tests/CMakeLists.txt

```diff
@@ -30,6 +30,23 @@ target_link_libraries(models_test
 # Get the list of sources from the test target.
 get_target_property(test_sources models_test SOURCES)
 
+# Go through the list of test sources and parse the test suite name.
```

Comment:
Not sure I get the incentive of the following here, why do we parse the test cases?

Reply:
Hey I asked the same question from Ryan and here is the answer from him that he wrote on the previous PR: 
Essentially this part of the code allows CTest integration for each model type. So, with this code, we have the ability to have one CTest test suite for each model type, which is nice. Otherwise, if you used CTest, it groups everything as one test suite, but especially for these models, tests for a model type may take a long time and so it might be preferable to have the ability to run only one test at a time.

Reply:
Ahh, I see, yes that only applies to ctest if you just use the main file you can use the catch command line arguments to run specific tests.

Reply:
Btw. you can use cmake message - https://cmake.org/cmake/help/latest/command/message.html to print out some results, e.g. to check if the regex works as expected.

Reply:
Yup i thought about the same that catch provides a list if tests on it's own and why do we need to do it through ctest but then i thought that maybe ctest is like the standard way of testing. If it's not necessary we can keep it to that because i am still having problems figuring this out.

Reply:
Also i will take a look at those ctest messages and see if i can make it work.

Reply:
Hey I added the parsing for ctest there was an easy solution provided by catch itself and i used that to add tests for ctest.
cc @rcurtin, @kartikdutt18